### PR TITLE
RD-5371 Fix support for `placeHolder` widget configuration parameter

### PR DESCRIPTION
**Pull request by qooban**
## Description
Fixes RD-5371 by passing `placeHolder` configuration parameter to `GenericField` component (b62517b781ba9f6fbb312d7c1676ef8dde5fd927).
In addition migrated few components from `EditWidget*` family to TypeScript (f4e0ae30121a2c37745b6a227f4710fc1bcf070d).

## Screenshots / Videos

image

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3149/) (15.07.2022) - all PASSED

## Documentation
N/A

### CODE REVIEW

File: app/utils/Toolbox.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ const createToolbox = (store: Store<ReduxState>) => {
 const getToolbox = (
     onRefresh: Stage.Types.Toolbox['refresh'],
     onLoading: Stage.Types.Toolbox['loading'],
-    widget: ReturnType<Stage.Types.Toolbox['getWidget']>
+    widget?: ReturnType<Stage.Types.Toolbox['getWidget']>
```
Comment (NorbertAmartus) [inline app/utils/Toolbox.ts:132]: Why it was modified to be optional? 🤔

Comment (qooban) [reply app/utils/Toolbox.ts:132]: For widgetless toolbox is not necessary to be passed.

Comment (NorbertAmartus) [reply app/utils/Toolbox.ts:132]: Okay, thanks for the explaination 🎉

**Review by NorbertAmartus — APPROVED** [review]
Looks amazing! 🥳 
Thanks for putting additionall effort on fully migrating files to TS! 🎉 

In that phase, 100% TS migration will happen after 3/4 of your PRs 😄 
Great job! 🎸

**Pull request merged**